packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -370,7 +370,10 @@ export async function handleRoute({
 	// Apply the `status` override to the response object before responding.
 	// Response.status is read-only, so a clone is required to override.
 	if (status && response.status !== status && (status === 404 || status === 500)) {
-		response = new Response(response.body, { ...response, status });
+		response = new Response(response.body, {
+			status: status,
+			headers: response.headers
+		});
```

Comment:
Can confirm that `...response` seems to do nothing here 👍 We could also copy over `statusText`, but not a big deal I think

Reply:
That was it originally. I advised against it.

https://github.com/withastro/astro/pull/9642#discussion_r1444741971

Reply:
Ah, looks like you're already one step ahead 😄